package Location;


message mac_address
{
	optional bytes addr = 1;
}

enum target_dev_type
{
	TARGET_TYPE_UNKNOWN          = 0;
	TARGET_TYPE_STATION          = 1;
	TARGET_TYPE_TAG              = 2;
	TARGET_TYPE_ROGUE            = 3;
}

enum algorithm
{
    ALGORITHM_TRIANGULATION = 0;
    ALGORITHM_AP_PLACEMENT = 1;
    ALGORITHM_CALIBRATION = 2;
    ALGORITHM_ESTIMATION = 3;
    ALGORITHM_LOW_DENSITY = 4;
}

enum measurement_unit
{
    METERS = 0;
    FEET = 1;
    PIXELS = 2;
}

message record {
	required uint32 timestamp = 1;
	required mac_address radio_mac = 2;
	required int32 rssi_val = 3;
	optional uint32 channel = 4;
}

enum error_code {
	ERROR_CODE_NO_ERROR = 0; // No error (a new fix is available)
	ERROR_CODE_0_RSSI = 1; // 0 RSSI could be used with the PDB
	ERROR_CODE_ONLY_1_RSSI = 2; // 1 RSSI could be used with the PDB
	ERROR_CODE_ONLY_2_RSSI = 3; // 2 RSSI could be used with the PDB
	ERROR_CODE_RSSI_QUALITY = 4; // RSSIs measurement are poor quality
	ERROR_CODE_RSSI_OLD_TIMESTAMP = 8; // timestamps of RSSI is too old
	ERROR_CODE_RSSI_CLOSE_TIMESTAMP = 16; // time since last fix computed is under threshold
	ERROR_CODE_LEGACY = 0xFFFFF; // This is the legacy error code for ALE version < 2.0.0.15 (internal usage)
}

message stream_location
{
	optional float sta_location_x               = 1;
	optional float sta_location_y               = 2;
	optional uint32 error_level                 = 3;
	optional string hashed_sta_eth_mac          = 7;
	repeated string geofence_ids                = 8;
	optional algorithm loc_algorithm            = 9;
	optional double longitude                   = 11;
	optional double latitude                    = 12;
	optional double altitude                    = 13;
	optional measurement_unit unit              = 14;
	required mac_address sta_eth_mac            = 15;
	optional string campus_id_string            = 16;
	optional string building_id_string          = 17;
	optional string floor_id_string             = 18;
	optional target_dev_type target_type        = 19 [default=TARGET_TYPE_UNKNOWN];
    optional bool associated                    = 20;
	optional error_code err_code                = 21;
	repeated record records                     = 22;
}

enum zone_event {
	ZONE_IN = 0;
	ZONE_OUT = 1;
}

message stream_geofence_notify {
	optional zone_event geofence_event = 1;
	optional bytes geofence_id = 2;
	optional string geofence_name = 3;
	optional mac_address sta_eth_mac = 4;
	optional bool associated = 5;
	optional uint32 dwell_time = 6 [default=0];
	optional string hashed_sta_eth_mac = 7;
}
